massa-node/src/main.rs

```diff
@@ -189,6 +189,8 @@ async fn launch(
     } else if SETTINGS.ledger.disk_ledger_path.exists() {
         std::fs::remove_dir_all(SETTINGS.ledger.disk_ledger_path.clone())
             .expect("disk ledger delete failed");
+        std::fs::remove_dir_all(SETTINGS.execution.hd_cache_path.clone())
```

Comment:
We need to make sure this doesn't happen when we reboot the reference node from the data on disk

Reply:
This code is in a else if. If it's a keep_ledger node it will enter in the if above.